### PR TITLE
Feat/1414 add error tracking & feedback

### PR DESCRIPTION
**Pull request by luxumbra**
## Overview

**What features/fixes does this PR include?**
- Adds Honeybadger error tracking to the frontend and is also using a hook to post updates to the [#-🦡-honeybadger-reports](https://discord.com/channels/629411177947987986/1030538751279104110) in Discord. 
- Also adds [UserBack](https://userback.io) for handling user feedback and bug reports. WRT privacy concerns, while we look into where we stand with and how to handle user data, we can turn off features/fields that use/request this info. I do think that giving a user the option to add their email is important as it allows us to follow up if needed.
- HotJar & Clarity are not included after @alalonde comments in #1414.

 

**Please provide the GitHub issue number**
#1414 

## Follow up Improvement Ideas

- [ ] Add to the backend?

## Implementation
Currently tested on local using env vars for setting `debug` &/or `reportData` `true` to send data to Honeybadger. 
[Honeybadger Docs for React](https://docs.honeybadger.io/lib/javascript/integration/react/)

**Describe technical (nontrivial / non-obvious) parts of your code**
All pretty self explanatory. I have gone through the frontend package and added `Honeybadger.notify(error)` to most places and enabled `enableUncaught` to capture uncaught errors/exceptions.
The public dashboard for errors (that have been shared) is here: [Public Dashboard](https://app.honeybadger.io/project/LWn2d4Bj1b?q=)

Ping me for any API keys,etc.

The UserBack config is done via their dashboard.

**Side effects**
Better visibility of what our frontend is doing in the wild. 

## Assets
image

### CODE REVIEW

File: packages/web/components/ConnectedPage.tsx [hunk 1]
```diff
@@ -39,6 +40,7 @@ export const ConnectedPage: React.FC<{
   }
 
   if (error) {
+    Honeybadger.notify(error);
```
Comment (alalonde) [inline packages/web/components/ConnectedPage.tsx:43]: I think the right way to do this would be to create a generic errorHandler function in web/utils. Then import Honeybadger in that function (_in one place_). So if we want to switch to another error handling service we only have to change the code in one place.

File: packages/web/config.ts [hunk 2]
```diff
@@ -17,7 +17,13 @@ export const CONFIG = {
...
   publicURL: process.env.NEXT_PUBLIC_FRONTEND_URL,
   gaId: process.env.NEXT_PUBLIC_GA4_ID,
-  clarityId: process.env.NEXT_CLARITY_ID,
+  honeybadgerApiKey: process.env.NEXT_PUBLIC_HONEYBADGER_API_KEY,
+  honeybadgerReportData:
+    process.env.NEXT_PUBLIC_HONEYBADGER_REPORT_DATA === 'true',
+  honeybadgerDebug: process.env.NEXT_PUBLIC_HONEYBADGER_DEBUG === 'true',
+  userbackToken:
+    process.env.NEXT_PUBLIC_USERBACK_TOKEN ||
+    '37882|74240|mjn1zXGiYaozZR6qmqFfjl9Sk',
```
Comment (alalonde) [inline packages/web/config.ts:26]: These will need to be added to the dockerfile and github action as well I believe

File: packages/web/next-env.d.ts [hunk 3]
```diff
@@ -4,3 +4,5 @@ declare module '*.png';
 declare module '*.jpg';
 declare module '*.jpeg';
 declare module '*.svg';
+declare module '@honeybadger-io/webpack';
+declare module '@honeybadger-io/js';
```
Comment (alalonde) [inline packages/web/next-env.d.ts:8]: Why is this necessary?

File: packages/web/config.ts (hunk 2 above)
Comment (luxumbra) [reply packages/web/config.ts:26]: Correct. I did post in builders guild but a better place to mention it would have been in this PR. 😅

File: packages/web/next-env.d.ts (hunk 3 above)
Comment (luxumbra) [reply packages/web/next-env.d.ts:8]: It seemed to fix an issue i was having when running my local build but I've removed them now and all is fine - I think the issue was my environment node version had changed from 16.13.1  to stable.

File: packages/web/components/ConnectedPage.tsx (hunk 1 above)
Comment (luxumbra) [reply packages/web/components/ConnectedPage.tsx:43]: > I think the right way to do this would be to create a generic errorHandler function in web/utils. Then import Honeybadger in that function (_in one place_). So if we want to switch to another error handling service we only have to change the code in one place.

just checking - leaving the `HoneybadgerErrorBoundary` in `_app.tsx` is fine?

Comment (alalonde) [reply packages/web/components/ConnectedPage.tsx:43]: > just checking - leaving the `HoneybadgerErrorBoundary` in `_app.tsx` is fine?

Yup!

File: packages/web/config.ts (hunk 2 above)
Comment (alalonde) [reply packages/web/config.ts:26]: I think adding these to the `docker/frontend/Dockerfile` should be part of this PR. You could then build the container locally and see if the variables make it in properly. LMK if you need help with that, some trial and error is probably involved. If it works in the PR deployment we should be good

Comment (alalonde) [reply packages/web/config.ts:26]: no, these would go in the frontend container which is not included in the docker-compose. So you'd have to manually `docker build` it. That only gets you part of the way anyway, probably the best thing to do would be to just push what you have. The test is if those env variables end up in the PR deployment on Google cloud

Comment (luxumbra) [reply packages/web/config.ts:26]: > probably the best thing to do would be to just push what you have. 

Pushing now.

File: packages/web/config.ts [hunk 4]
```diff
@@ -17,7 +17,15 @@ export const CONFIG = {
     process.env.NEXT_BRIGHTID_NODE_URL || 'http:%2f%2fnode.brightid.org',
   publicURL: process.env.NEXT_PUBLIC_FRONTEND_URL,
   gaId: process.env.NEXT_PUBLIC_GA4_ID,
-  clarityId: process.env.NEXT_CLARITY_ID,
+  honeybadgerApiKey:
+    process.env.NEXT_PUBLIC_HONEYBADGER_API_KEY ||
+    'hbp_Z4027J7qYTktQkqvUHcOHJqIbpjpWD30A1kd',
```
Comment (alalonde) [inline packages/web/config.ts:22]: Should these be secrets?

**Comment by alalonde** [comment]
I submitted feedback through the userback widget, did it go anywhere? Any ideas on how to trigger a honeybader error/report on that PR env?

**Comment by luxumbra** [comment]
> I submitted feedback through the userback widget, did it go anywhere? Any ideas on how to trigger a honeybader error/report on that PR env?

Feedback received. Wrt triggering honeybadger, there's errors in the profile screen when editing & saving contents of blocks on a new account. They're handled already with toasts but should trigger a HB notify event.

File: packages/web/config.ts (hunk 4 above)
Comment (luxumbra) [reply packages/web/config.ts:22]: Probably should be. I'll change the keys and remove from config.

Comment (luxumbra) [reply packages/web/config.ts:22]: > Should these be secrets?

Do you have access to the cloud account?

Comment (alalonde) [reply packages/web/config.ts:22]: no

**Review by alalonde — APPROVED** [review]
